### PR TITLE
Add toAssoc Macro to Collections

### PR DESCRIPTION
**Pull request by akiyamaSM**
It transform a pair of [key, value] into a collection of [ key => value ]

Example

```php
$assoc = collect([
            ['name', 'John'],
            ['lastName', 'Doe'],
])->toAssoc();

$assoc->all(); // returns ['name' => 'John', 'lastName' => 'Doe']
```

### CODE REVIEW

File: tests/HelpersTest.php [hunk 1]
```diff
@@ -149,6 +149,22 @@ function stopwatch()
...
 
+    /** @test */
+    public function toAssoc()
+    {
+        $ordered = collect([
+            ['name', 'John'],
+            ['lastName', 'Doe'],
+        ])->toAssoc();
+
+        $this->assertTrue(
```
Comment (peter279k) [inline tests/HelpersTest.php]: How about using the `assertSame` to assert the expected value is same as `0` value?

File: src/CollectionMacros.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+namespace Calebporzio\AwesomeHelpers;
+
+
+class CollectionMacros
+{
+    public function toAssoc()
+    {
+        return function (){
```
Comment (peter279k) [inline src/CollectionMacros.php]: I think we should add the white space before the `(` character because of consistency.

**Comment by akiyamaSM** [comment]
Hello @calebporzio :D

**Comment by calebporzio** [comment]
I'd like to see some compelling use cases for this addition. I don't want this package to become a giant glob of things only a few people use. Thanks for being willing to contribute though.

**Pull request closed without merging**